apisix/plugins/openid-connect.lua

```diff
@@ -315,6 +324,24 @@ local function add_access_token_header(ctx, conf, token)
     end
 end
 
+-- Function to split the scope string into a table
+local function split_scopes_by_space(scope_string)
+    local scopes = {}
+    for scope in string.gmatch(scope_string, "%S+") do
+        scopes[scope] = true
+    end
+    return scopes
+end
+
+-- Function to check if all required scopes are present
+local function required_scopes_present(required_scopes, http_scopes)
```

Comment:
you can use `core.table.array_find` directly: https://github.com/apache/apisix/blob/master/apisix/core/table.lua#L94

Reply:
array_find(arr,ele) checks if `ele` is present in `arr`.
This function checks if all the elements in first table are present in second table.